cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -105,68 +112,77 @@ public CruiseControlApiImpl(Vertx vertx, int idleTimeout) {
-                    response.bodyHandler(buffer -> {
-                        String userTaskID = response.getHeader(CC_REST_API_USER_ID_HEADER);
-                        JsonObject json = buffer.toJsonObject();
-                        CruiseControlRebalanceResponse ccResponse = new CruiseControlRebalanceResponse(userTaskID, json);
-                        result.complete(ccResponse);
-                    });
-                } else if (response.statusCode() == 202) {
-                    response.bodyHandler(buffer -> {
-                        String userTaskID = response.getHeader(CC_REST_API_USER_ID_HEADER);
-                        JsonObject json = buffer.toJsonObject();
-                        CruiseControlRebalanceResponse ccResponse = new CruiseControlRebalanceResponse(userTaskID, json);
-                        if (json.containsKey(CC_REST_API_PROGRESS_KEY)) {
-                            // If the response contains a "progress" key then the rebalance proposal has not yet completed processing
-                            ccResponse.setProposalStillCalaculating(true);
-                        } else {
-                            result.fail(new CruiseControlRestException(
-                                    "Error for request: " + host + ":" + port + path +
-                                            ". 202 Status code did not contain progress key. Server returned: " +
-                                            ccResponse.getJson().toString()));
-                        }
-                        result.complete(ccResponse);
-                    });
-                } else if (response.statusCode() == 500) {
-                    response.bodyHandler(buffer -> {
-                        String userTaskID = response.getHeader(CC_REST_API_USER_ID_HEADER);
-                        JsonObject json = buffer.toJsonObject();
-                        if (json.containsKey(CC_REST_API_ERROR_KEY)) {
-                            // If there was a client side error, check whether it was due to not enough data being available
-                            if (json.getString(CC_REST_API_ERROR_KEY).contains("NotEnoughValidWindowsException")) {
-                                CruiseControlRebalanceResponse ccResponse = new CruiseControlRebalanceResponse(userTaskID, json);
-                                ccResponse.setNotEnoughDataForProposal(true);
-                                result.complete(ccResponse);
+            httpClient.request(HttpMethod.POST, port, host, path, request -> {
+                if (request.succeeded()) {
+                    if (idleTimeout != HTTP_DEFAULT_IDLE_TIMEOUT_SECONDS) {
+                        request.result().setTimeout(idleTimeout * 1000);
+                    }
+
+                    if (userTaskId != null) {
+                        request.result().putHeader(CC_REST_API_USER_ID_HEADER, userTaskId);
+                    }
+                    request.result().end();
```

Comment:
Doesn't `.end()` send the request? The [javadoc](https://vertx.io/docs/apidocs/io/vertx/core/http/HttpClientRequest.html#end--) says that "Once the request has ended, it cannot be used any more,"?

Reply:
Good catch.

Reply:
From this https://access.redhat.com/documentation/en-us/red_hat_build_of_eclipse_vert.x/4.0/pdf/eclipse_vert.x_4.0_migration_guide/Red_Hat_build_of_Eclipse_Vert.x-4.0-Eclipse_Vert.x_4.0_Migration_Guide-en-US.pdf
 I understand the `.end()` is not needed anymore.